pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -47,7 +53,7 @@ pub struct WalConnectionStatus {
 /// Open a connection to the given safekeeper and receive WAL, sending back progress
 /// messages as we go.
 pub async fn handle_walreceiver_connection(
-    id: ZTenantTimelineId,
+    timeline: Arc<Timeline>,
```

Comment:
As an idea for the future, we could send `Weak<Timeline>` into such places, requiring every operation to check for the timeline to be present.

That would require us to become quite rigorous on the timeline management side, so definitely out of scope of this PR.